### PR TITLE
Add org.mersenne.mprime (prime95)

### PR DESCRIPTION
**Pull request by sat3ll**
Request for review for adding prime95/mprime.
Uses: record prime finding, stress testing.
Upstream/developer has acknowledged the existence of this.

About the daemon launching script: mprime/prime95, when used for its primary activity (prime hunting), is seldom useful if not launched as a daemon. Leaving this option in would benefit most users.

### CODE REVIEW

**Comment by nedrichards** [comment]
bot, build org.mersenne.mprime

File: README.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# Installation
```
Comment (barthalion) [inline README.md]: I don't think it's a good fit for Flathub if you expect user to create systemd unit. Instructions won't also cover all possible options: what if user doesn't use systemd? Also README duplicates official flatpak docs.

File: org.mersenne.mprime.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>org.mersenne.mprime</id>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>EULA</project_license>
```
Comment (barthalion) [inline org.mersenne.mprime.appdata.xml]: Does the license allow redistribution?

File: org.mersenne.mprime.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+runtime: org.freedesktop.Platform
+runtime-version: '18.08'
+sdk: org.freedesktop.Sdk
+
+command: mprime-wrapper
+tags:
+  - proprietary
+
+finish-args:
+  - --share=network
```
Comment (barthalion) [inline org.mersenne.mprime.yml]: What does it use network for?

Comment (sat3ll) [reply org.mersenne.mprime.yml]: It's a distributed computing program, it uses network to fetch "work units".

File: org.mersenne.mprime.appdata.xml (hunk 2 above)
Comment (sat3ll) [reply org.mersenne.mprime.appdata.xml]: Skimming through the EULA, I don't see any immediate clause regarding redistribution, although I've already contacted the developer of it and the only requirement to meet before submitting it here was to make sure to update it to version 29.8 (current stable one, was a beta version at time of submission).

File: README.md (hunk 1 above)
Comment (sat3ll) [reply README.md]: I wouldn't say I expect them to create a systemd unit. The "default" launching behavior here is to start it in interactive mode, which should cover the most common usage, which is stress-testing.
The interactive mode also allows it to be used for distributed computing but it's only in this case that the systemd unit becomes useful.
About the possible options for init systems, unfortunately I've got no idea/experience about the others. So far, all my systems have used systemd.

**Comment by nedrichards** [comment]
bot, build org.mersenne.mprime

**Comment by nedrichards** [comment]
We've just enabled automated appdata verification so there's some exciting new error messages:
```
flatpak run org.freedesktop.appstream-glib validate builddir/*/share/appdata/org.mersenne.mprime.appdata.xml
 in dir /srv/buildbot/worker/build-x86_64-4/build (timeout 1200 secs)
 watching logfiles {}
 argv: b'flatpak run org.freedesktop.appstream-glib validate builddir/*/share/appdata/org.mersenne.mprime.appdata.xml'
 using PTY: False
builddir/files/share/appdata/org.mersenne.mprime.appdata.xml: FAILED:
• tag-invalid           : <project_license> is not valid [EULA]SPDX ID 'EULA' unknown
• tag-missing           : <translation> not specified
• tag-missing           : <update_contact> is not present
• style-invalid         : Not enough <screenshot> tags, minimum is 1
• style-invalid         : <p> cannot contain a hyperlink [This program is used to find Mersenne Prime numbers.  See http://primes.utm.edu/mersenne/index.html for a good description of Mersenne primes.  Mersenne numbers can be proved composite (not prime) by either finding a factor or by running a Lucas-Lehmer primality test.]
Validation of files failed
program finished with exit code 1
elapsedTime=0.152257
```

File: org.mersenne.mprime.appdata.xml (hunk 2 above)
Comment (nedrichards) [reply org.mersenne.mprime.appdata.xml]: This isn't valid appdata/spdx. It needs to be `LicenseRef-proprietary`

**Comment by LeandroStanger** [comment]
bot, build org.mersenne.mprime

**Comment by sat3ll** [comment]
> We've just enabled automated appdata verification so there's some exciting new error messages:

Does the appstream metadata file now need to pass **validate** instead of **validate-relax**?

**Comment by nedrichards** [comment]
@sat3ll it's supposed to be validate-relax plus the bits that the flathub guidelines said, it seems that it's slightly more restricted than intended. I believe @barthalion will be working on removing some of the bits but in your case you should be able to put an empty translation tag in there and be fine (if there aren't any translations)

**Comment by sat3ll** [comment]
bot, build org.mersenne.mprime

**Comment by barthalion** [comment]
bot, build org.mersenne.mprime

**Comment by barthalion** [comment]
Repository has been created: https://github.com/flathub/org.mersenne.mprime

Let us know if someone from upstream team should also have write access there. Thanks!

**Pull request closed without merging**